tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -162,7 +165,8 @@ private[scalanative] object LLVM {
       }
       linkerResult.links.map(_.name) ++ librt ++ libunwind ++ config.gc.links
     }
-    val linkopts = config.linkingOptions ++ links.map("-l" + _) ++ Seq(
+    val osxOpts = if (Properties.isMac) Seq("-framework", "Foundation") else Nil
+    val linkopts = config.linkingOptions ++ links.map("-l" + _) ++ osxOpts ++ Seq(
```

Comment:
It looks like we go from two dynamic dependencies for hello world:
```
# otool -L sandbox/target/scala-2.11/sandbox-out 
sandbox/target/scala-2.11/sandbox-out:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)
	/usr/local/opt/llvm/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1.0.0)
```
To four after this change:
```
# otool -L sandbox/target/scala-2.11/sandbox-out 
sandbox/target/scala-2.11/sandbox-out:
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1349.91.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)
	/usr/local/opt/llvm/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
```
Given our long-term strive to minimize number of dynamic dependencies (#1139), I don't think supporting macOS version string justifies the added cost. 

I believe that reporting empty string for version on macOS is a better alternative.

Reply:
I agree that minimizing dynamic dependencies is a great idea. That being said, the two new libraries are not really external dependencies. They are system built-ins that as far as I can tell are guaranteed to be present on any apple platform: https://developer.apple.com/documentation/foundation?changes=_5&language=objc.

Also, I didn't really make this clear in the commit, but I was actually more concerned with the temp directory location than the os version string. I don't think there is any way other than using the framework api to get the correct user temp directory location. Of course, this is probably not a huge deal, `/tmp` still works, but it is not where Files.createTempDirectory puts the new temp directories on the jvm.